src/Engine.cpp

```diff
@@ -607,11 +607,62 @@ static void opt_version() {
 		"...please don't sue us." << std::endl;
 }
 
-bool Engine::parseCommandLine(int argc, char *argv[]) {
+bool Engine::parseCommandLine(int argc, char* argv[])
+{
+
 	// variables for command-line flags
 	int optret;
 	std::vector<std::string> data_vec;
 
+
+
+#ifdef _WIN32
+
+	char path[MAX_PATH];
+
+	HRESULT hr = CoInitializeEx(NULL, COINIT_APARTMENTTHREADED |
+		COINIT_DISABLE_OLE1DDE);
+	IFileOpenDialog* fo;
+
+	hr = CoCreateInstance(CLSID_FileOpenDialog, NULL, CLSCTX_ALL,
+		IID_IFileOpenDialog, reinterpret_cast<void**>(&fo));
+
+	fo->SetOptions(FOS_PICKFOLDERS);
+
+	if (SUCCEEDED(hr))
+	{
+		// Show the Open dialog box.
+		hr = fo->Show(NULL);
+
+		// Get the file name from the dialog box.
+		if (SUCCEEDED(hr))
+		{
+			IShellItem* pItem;
+			hr = fo->GetResult(&pItem);
+			if (SUCCEEDED(hr))
+			{
+				PWSTR pszFilePath;
+				hr = pItem->GetDisplayName(SIGDN_FILESYSPATH, &pszFilePath);
+				wcstombs(path, pszFilePath, MAX_PATH);
+				data_vec.push_back(path);
+			}
+		}
+		else
+		{
+			std::cout << "Couldn't show File Open Dialog" << std::endl;
+		}
+		fo->Release();
+
+		CoUninitialize();
```

Comment:
Make sure this is always called if CoInitializeEx succeeds. We don't have a scope guard class in here, but you could do:

```c++
hr = CoInitializeEx(...)
If (!SUCCESS(hr)) return;
struct coinit_scope {
  ~coinit_scope() { CoUninitialize(); }
} coinit_scope_instance;
```